tests/test_unittest/test_gtscript_frontend.py

```diff
@@ -240,6 +240,103 @@ def definition_func(inout_field: gtscript.Field[float]):
                 definition_func, name=inspect.stack()[0][3], module=self.__class__.__name__
             )
 
+    def test_use_in_expr(self):
+        @gtscript.function
+        def func():
+            return 1.0
+
+        def definition_func(inout_field: gtscript.Field[float]):
+            from gt4py.__gtscript__ import PARALLEL, computation, interval
+
+            with computation(PARALLEL), interval(...):
+                inout_field = func() + 1
+
+        parse_definition(
+            definition_func, name=inspect.stack()[0][3], module=self.__class__.__name__
+        )
+
+    def test_use_as_func_arg(self):
+        @gtscript.function
+        def func():
+            return 1.0
+
+        @gtscript.function
+        def func_outer(arg):
+            return arg + 1
+
+        def definition_func(inout_field: gtscript.Field[float]):
+            from gt4py.__gtscript__ import PARALLEL, computation, interval
+
+            with computation(PARALLEL), interval(...):
+                inout_field = func_outer(func())
+
+        parse_definition(
+            definition_func, name=inspect.stack()[0][3], module=self.__class__.__name__
+        )
+
+    def test_use_in_expr_in_func_arg(self):
+        @gtscript.function
+        def func():
+            return 1.0
+
+        @gtscript.function
+        def func_outer(arg):
+            return arg + 1
+
+        def definition_func(inout_field: gtscript.Field[float]):
+            from gt4py.__gtscript__ import PARALLEL, computation, interval
+
+            with computation(PARALLEL), interval(...):
+                inout_field = func_outer(func() + 1)
+
+        parse_definition(
+            definition_func, name=inspect.stack()[0][3], module=self.__class__.__name__
+        )
```

Comment:
Have you verified that this works when `func` has a few statements instead of just a return? I realize the tests below sort of do this, but they raise for other reasons so it might be a blind spot.

Reply:
I think these issues are not related to what this PR: We do not implement function calls here, rather that they are dealt with in the context of expressions. Which should e.g. not be affected by the complexity of the function. I'd propose opening a separate issue if there should be more test cases for functions in general.

Reply:
Sorry I probably wasn't clear. What I meant was that there is some tricky logic that unrolls functions and inserts them before the statement with the call. I hope that the `CallInliner` logic still works when the function call is embedded in a Call.

Your tests probably test that sufficiently, but it was just a thought as I went through.

Reply:
Oh yes, the call inliner works recursively.